### PR TITLE
Add support for LLVM 17

### PR DESCRIPTION
**Pull request by rjaegers**
This PR adds support to build Mull with LLVM 17.

I have not yet updated any documentation or ci infrastructure. Please see if this can be incorporated into Mull, if any further modifications are required I can do so. This PR supports our [devcontainer](https://github.com/philips-software/amp-devcontainer) use-case.

### CODE REVIEW

**Review by AlexDenisov — APPROVED** [review]
Thank you so much for submitting this PR, highly appreciated!

**Comment by AlexDenisov** [comment]
There are some failures unrelated to your changes, I think I need to update our ansible playbooks, I'll try to get to it by the end of this week.

**Comment by AlexDenisov** [comment]
@rjaegers I fixed the CI issue, could you please update the PR against `main`? Happy to merge it as soon as CI is green!

File: tests/Helpers/InMemoryCompiler.cpp [hunk 1]
```diff
@@ -52,7 +52,7 @@ std::unique_ptr<llvm::Module> InMemoryCompiler::compile(const std::string &code,
   auto &headerSearchOptions = compilerInvocation.getHeaderSearchOpts();
   headerSearchOptions.Verbose = false;
   auto &codeGenOptions = compilerInvocation.getCodeGenOpts();
-  codeGenOptions.setDebugInfo(clang::codegenoptions::FullDebugInfo);
+  codeGenOptions.setDebugInfo(llvm::codegenoptions::FullDebugInfo);
```
[the remark this answers is not in the document]
Comment (rjaegers) [reply tests/Helpers/InMemoryCompiler.cpp]: Indeed, it did :).

**Comment by rjaegers** [comment]
@AlexDenisov updated with latest changes from main and fixed an issue with the tests for LLVM < 17. I think it should be good to go.

Happy to do something back for this awesome project!

**Pull request merged**